src/nifti1.ts

```diff
@@ -0,0 +1,617 @@
+      ", " +
+      fmt(this.affine[2][3]) +
+      "\n";
+
+    string += 'Intent Name: "' + this.intent_name + '"\n';
+
+    if (this.extensionFlag[0]) {
+      string +=
+        "Extension: Size = " +
+        this.extensionSize +
+        "  Code = " +
+        this.extensionCode +
+        "\n";
+    }
+
+    return string;
+  }
+
+  /**
+   * Returns the byte index of the extension.
+   * @returns {number}
+   */
+  protected getExtensionLocation(): number {
+    return NIFTI1.MAGIC_COOKIE + 4;
+  }
+
+  /**
+   * Returns header as ArrayBuffer.
+   * @param {boolean} includeExtensions - should extension bytes be included
+   * @returns {ArrayBuffer}
+   */
+  public toArrayBuffer(includeExtensions = false): ArrayBuffer {
+    const SHORT_SIZE = 2;
+    const FLOAT32_SIZE = 4;
+    let byteSize = 348 + 4; // + 4 for the extension bytes
+
+    // calculate necessary size
+    if (includeExtensions) {
+      for (let extension of this.extensions) {
+        byteSize += extension.esize;
+      }
+    }
+    let byteArray = new Uint8Array(byteSize);
+    let view = new DataView(byteArray.buffer);
+    // sizeof_hdr
+    view.setInt32(0, 348, this.littleEndian);
+
+    // data_type, db_name, extents, session_error, regular are not used
+
+    // dim_info
+    view.setUint8(39, this.dim_info);
+
+    // dims
+    for (let i = 0; i < 8; i++) {
+      view.setUint16(40 + SHORT_SIZE * i, this.dims[i]!, this.littleEndian);
+    }
+
+    // intent_p1, intent_p2, intent_p3
+    view.setFloat32(56, this.intent_p1, this.littleEndian);
+    view.setFloat32(60, this.intent_p2, this.littleEndian);
+    view.setFloat32(64, this.intent_p3, this.littleEndian);
+
+    // intent_code, datatype, bitpix, slice_start
+    view.setInt16(68, this.intent_code, this.littleEndian);
+    view.setInt16(70, this.datatypeCode, this.littleEndian);
+    view.setInt16(72, this.numBitsPerVoxel, this.littleEndian);
+    view.setInt16(74, this.slice_start, this.littleEndian);
+
+    // pixdim[8], vox_offset, scl_slope, scl_inter
+    for (let i = 0; i < 8; i++) {
+      view.setFloat32(
+        76 + FLOAT32_SIZE * i,
+        this.pixDims[i]!,
+        this.littleEndian
+      );
+    }
+    view.setFloat32(108, this.vox_offset, this.littleEndian);
+    view.setFloat32(112, this.scl_slope, this.littleEndian);
+    view.setFloat32(116, this.scl_inter, this.littleEndian);
+
+    // slice_end
+    view.setInt16(120, this.slice_end, this.littleEndian);
+
+    // slice_code, xyzt_units
+    view.setUint8(122, this.slice_code);
+    view.setUint8(123, this.xyzt_units);
+
+    // cal_max, cal_min, slice_duration, toffset
+    view.setFloat32(124, this.cal_max, this.littleEndian);
+    view.setFloat32(128, this.cal_min, this.littleEndian);
+    view.setFloat32(132, this.slice_duration, this.littleEndian);
+    view.setFloat32(136, this.toffset, this.littleEndian);
+
+    // glmax, glmin are unused
+
+    // descrip and aux_file
+    byteArray.set(Buffer.from(this.description), 148);
```

Comment:
this works in tests since they run under node.js. However, typescript is highlighting the browsers don't have `Buffer`.  Should write a minimal polyfill for `Buffer.from` since that appears to be the only usage (`string` => `Uint8Array`)